### PR TITLE
Fix proving opts

### PR DESCRIPTION
**Pull request by brusherru**
No issue. Critical thing!

### CODE REVIEW

File: desktop/main/smeshingOpts.ts [hunk 1]
```diff
@@ -47,6 +54,27 @@ export const isSmeshingOpts = (a: any): a is SmeshingOpts =>
...
+    nonces > 0 &&
+    typeof threads === 'number' &&
+    threads > 0
+  );
+};
+
+export const safeProvingOpts = (
+  opts?: Partial<SmeshingProvingOpts>
+): SmeshingProvingOpts | Record<string, never> =>
+  isValidProvingOpts(opts) ? opts : {};
```
Comment (pigmej) [inline desktop/main/smeshingOpts.ts:74]: So if I read that part correctly, we either use non zeros or empty and then default to wherever is in go-sm?

Comment (maparr) [reply desktop/main/smeshingOpts.ts:74]: if empty or zero, we use the default, otherwise, we use values from the config.

**Comment by monikasmolarek** [comment]
If both smeshing proving opts were set to 0, the configs weren't corrected with the defaults, smeshing opts are now empty: 
Screenshot (3)

**Review by maparr — APPROVED** [review]

**Review by monikasmolarek — APPROVED** [review]

**Comment by brusherru** [comment]
> If both smeshing proving opts were set to 0, the configs weren't corrected with the defaults, smeshing opts are now empty:

Yes, that's correct behavior — we drop any invalid values or keep only valid :)

**Pull request merged**